app/validators/uri_validator.rb

```diff
@@ -4,16 +4,24 @@ class UriValidator < ActiveModel::EachValidator
   DEFAULT_ACCEPTED_SCHEMES = /^https?$/
   DEFAULT_OPTIONAL_PARTS = %i[port].freeze
   DEFAULT_FORBIDDEN_PARTS = %i[userinfo registry path opaque query fragment].freeze
+  MAX_HOST_SIZE = 255.freeze
+  MAX_LABEL_SIZE = 63.freeze
 
   def validate_each(record, attribute, value)
-    valid = begin
-              uri = URI.parse(value)
-              valid_scheme?(uri.scheme) && valid_host?(uri.host) && !forbidden_part?(record, uri)
-            rescue URI::InvalidURIError
-              false
-            end
-    record.errors.add(attribute, options[:message] || :invalid) unless valid
-    valid
+    uri = nil
+
+    begin
+      uri = URI.parse(value)
+    rescue URI::InvalidURIError
+      add_generic_failure_message(record, attribute) and return false
+    end
+
+    # It's done this way to get all the errors even if it is already known that is invalid
+    [
+      validate_scheme(record, attribute, uri.scheme),
```

Comment:
Wouldn't `uri` be `nil` when rescuing `URI::InvalidURIError`?

Reply:
@josemigallas yes, that is why it has a return before reaching here 😄